tests/aziot/aziot_test.go

```diff
@@ -0,0 +1,260 @@
+}
+
+func TestAzureIotTPMEndrolmentWithVTPM(t *testing.T) {
+	t.Log("TestAzureIotTPMEndrolmentWithVTPM started")
+	t.Log("Setup :\n\tAziot (latest) on Ubuntu-22.04-amd64\n\twith VTPM")
+	defer t.Log("TestAzureIotTPMEndrolmentWithVTPM finished")
+
+	t.Skip("Skip test for now, it is failing!")
+
+	testAzureIotEdge(t, "aziot-latest", true)
+}
+
+func testAzureIotEdge(t *testing.T, version string, useVTPM bool) {
+	appName := tk.GetRandomAppName(projectName)
+	pubPorts := []string{sshPort + ":22"}
+	pc := tk.GetDefaultVMConfig(appName, tk.AppDefaultCloudConfig, pubPorts)
+	err := eveNode.EveDeployApp(appLink[version], pc,
+		tk.WithSSH(tk.AppDefaultSSHUser, tk.AppDefaultSSHPass, sshPort))
+	if err != nil {
+		t.Fatalf("Failed to deploy app: %v", err)
+	}
+	defer func() {
+		err = eveNode.AppStopAndRemove(appName)
+		if err != nil {
+			log.Errorf("Failed to stop and remove app: %v", err)
+		}
+	}()
+
+	// wait for the deployed app to appear in the list
+	time.Sleep(30 * time.Second)
+
+	// wait maximum of 5 minutes for the app to start
+	t.Logf("Waiting for app %s to start...", appName)
+	err = eveNode.AppWaitForRunningState(appName, 60*5)
+	if err != nil {
+		t.Fatalf("Failed to wait for app to start: %v", err)
+	}
+
+	t.Logf("Waiting for ssh to be ready...")
+	err = eveNode.AppWaitForSSH(appName, 60*5)
+	if err != nil {
+		t.Fatalf("Failed to wait for ssh: %v", err)
+	}
+	t.Log("SSH connection established")
+
+	testScriptPath := testScriptBasePath + filepath.Base(testScript[version])
+	err = eveNode.AppSCPCopy(appName, testScript[version], testScriptPath)
+	if err != nil {
+		t.Fatalf("Failed to copy file to vm: %v", err)
+	}
+	t.Log("Test script copied to VM")
+
+	t.Log("Creating a TPM enrollment in Azure IoT Hub")
+	endorsementKey, enrollmentID := "", ""
+	if !useVTPM {
+		// for this to test to work, we need to create an enrollment in the Azure IoT Hub,
+		// the enrolment should be created with the endorsement key of the TPM and
+		// since we are running EVE in QEMU with SWTPM, the endorsement key changes
+		// every time we start the VM EVE, so we need to read it, create the enrollment,
+		// run the test and delete the enrollment.
+		ek, id, err := readEveEndorsmentKey()
+		if err != nil {
+			t.Fatalf("Failed to read endorsement key from EVE: %v", err)
+		}
+		endorsementKey, enrollmentID = ek, id
+	} else {
+		createKeyScriptPath := testScriptBasePath + "test_make_tpm_keys.sh"
+		err = eveNode.AppSCPCopy(appName, "scripts/test_make_tpm_keys.sh", createKeyScriptPath)
+		if err != nil {
+			t.Fatalf("Failed to copy file to vm: %v", err)
+		}
+
+		// prepare the script for execution
+		command := fmt.Sprintf("chmod +x %s", createKeyScriptPath)
+		_, err = eveNode.AppSSHExec(appName, command)
+		if err != nil {
+			t.Fatalf("Failed perpare the TPM key creation script for execution: %v", err)
+		}
+
+		// execute the script to create the necessary TPM keys
+		_, err = eveNode.AppSSHExec(appName, createKeyScriptPath)
```

Comment:
side note: I'd abstract path to a file into a reader for AppSSHExec function, it'll be more flexible to call and if you want to run some local examples (unit tests for a function) you can create readers in code :)

Reply:
you mean change AppSCPCopy to something like AppSCPCopy(strAppname, ioReader, strRemoteFile) ?
but the internal scp call needs a path.